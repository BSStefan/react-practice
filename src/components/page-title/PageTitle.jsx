// @flow
import * as React from 'react';
import styled from 'styled-components';
import { brand } from '../../styles/colors';

type Props = {
  title: string,
  as: string
}

const PageTitle = (props: Props) => {
  const { title, as } = props;
  return <Header as={as}>{title}</Header>;
};

PageTitle.defaultProps = {
  as: 'h1'
};

const Header = styled.h1`
  color: ${brand};
`;

export default PageTitle;