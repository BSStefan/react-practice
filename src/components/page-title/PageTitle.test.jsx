import * as React from 'react';
import * as renderer from 'react-test-renderer';
import PageTitle from './PageTitle';

const defaultProps = {
  title: 'Test title',
}

it('Should match snapshout with default h1', () => {
  const tree = renderer.create(<PageTitle {...defaultProps} />).toJSON();
  expect(tree).toMatchSnapshot();
})

it('Should match snapshout when h2 is sent', () => {
  const tree = renderer.create(<PageTitle {...defaultProps} as="h2" />).toJSON();
  expect(tree).toMatchSnapshot();
})