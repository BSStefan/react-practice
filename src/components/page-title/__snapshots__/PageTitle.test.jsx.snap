// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshout when h2 is sent 1`] = `
<h2
  className="sc-bdVaJa lnRlpM"
>
  Test title
</h2>
`;

exports[`Should match snapshout with default h1 1`] = `
<h1
  className="sc-bdVaJa lnRlpM"
>
  Test title
</h1>
`;
